framework/db/SchemaBuilder.php

```diff
@@ -0,0 +1,412 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\db;
+
+use Yii;
+use yii\base\Object;
+
+/**
+ * SchemaBuilder is the class help to define DB's schema types.
+ *
+ * For example you may use the following code inside your migration files:
+ *
+ * ```php
+ * $this->createTable('{{table}}', [
+ *   'name' => Schema::string(64)->notNull(),
+ *   'type' => Schema::integer()->notNull()->default(10),
+ *   'description' => Schema::text(),
+ *   'rule_name' => Schema::string(64),
+ *   'data' => Schema::text(),
+ *   'created_at' => Schema::integer(),
+ *   'updated_at' => Schema::integer(),
+ *]);
+ * ```
+ *
+ * @method SchemaBuilder default($default = null) see [[SchemaBuilder::_default()]] for more info
+ *
+ * @author Vasenin Matvey <vaseninm@gmail.com>
+ * @since 2.0.5
+ */
+abstract class SchemaBuilder extends Object
+{
+    /**
+     * @var string column schema
+     */
+    protected $schema = null;
+    /**
+     * @var integer column size
+     */
+    protected $length = null;
+    /**
+     * @var boolean whether value could be null
+     */
+    protected $isNotNull = null;
```

Comment:
what is the semantic of `null` here? a column is either nullable or not. The `is` prefix is also confusing. a column is not null itself, it is nullable.